packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -387,23 +387,123 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
     notifyListeners();
   }
 
-  void _debugAssertIsValidOrientation(ScrollbarOrientation orientation) {
-    assert(
-    (_isVertical && _isVerticalOrientation(orientation)) || (!_isVertical && !_isVerticalOrientation(orientation)),
-    'The given ScrollbarOrientation: $orientation is incompatible with the current AxisDirection: $_lastAxisDirection.'
+  // - Scrollbar Details
+
+  Rect? _trackRect;
+  double get _trackExtent => _lastMetrics!.viewportDimension - _totalMainAxisInsets;
+
+  Rect? _thumbRect;
+  late double _thumbOffset;
+  late double _thumbExtent;
+  void _setThumbExtent() {
+    // Thumb extent reflects fraction of content visible, as long as this
+    // isn't less than the absolute minimum size.
+    // extentInside <= viewportDimension and used for appropriate scaling
+    // accounting for insets & padding.
+    final double fractionVisible = clampDouble(
+      _lastMetrics!.extentInside / _totalContentExtent,
```

Comment:
I looked into making that change with the padding @xu-baolin and I found it is actually correct. The fractionVisible takes into account the padding because the viewport can extend behind widgets like app bars. There are some `scrollbar_paint_tests` that demonstrate this.
Say there is a cupertino navigation bar like in this test: https://github.com/flutter/flutter/blob/2d771d4966b6a8562bd026807fccbbddecf42e2e/packages/flutter/test/cupertino/scrollbar_paint_test.dart#L64

The `fractionVisible` needs to account for the navigation bar since the `viewportDimension` does not. 

This also reminded me of: https://github.com/flutter/flutter/blob/2d771d4966b6a8562bd026807fccbbddecf42e2e/packages/flutter/lib/src/widgets/scroll_view.dart#L662

That is the padding that was being considered here. I'll add it back to this PR with some comments. It really confused me for a minute, because you are so right. When you read it, it does not look right when computing fractionVisible, but it is. 

All of the other minor bugs have been fixed, so I will try to have this PR updated today and ready to land. It should not require updating any tests since it is meant to just be a clean up. 🤞  😜